ls1intum/Artemis
# `Assessment`: Fix issues in the scores view for multiple correction rounds and file upload exercises

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


Fix #6395
Additionally fix issue where unassessed file upload participations where not shown in the scores table.

### Description

The scores table now correctly displays the buttons for each assessment of different correction rounds in exams.
Additionally file upload exercises are correctly displayed and allow the redirection to their assessment view.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 File Upload exercise

1. Log in to Artemis
2. Participate as student
3. Verify that the buttons in the scores table still work and redirect the instructor to the correct assessment.

#### Exam Mode Testing



Prerequisites:
- 1 Instructor
- 1 Student
- 1 Exam with two correction rounds

1. Log in to Artemis
2. Participate in the exam as a student
3. Go to the scores table of an exercise and verify that the buttons correctly work in each of these three scenarios:
- Before all assessments
- After the assessment of the first correction round
- After the assessment in the second correction round

### Review Progress






#### Code Review
#### Manual in Exam mode Tests
#### Manual Test verifying that course exercises work

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| ParticipationResource.java | 69% | 91% |
| exercise-scores.component.ts.ts | 76% | 90% |

### Screenshots


<img width="1728" alt="Bildschirmfoto 2023-04-04 um 15 57 35" src="https://user-images.githubusercontent.com/38322605/229816246-3ca2a8c3-1262-4aae-b7a7-ab0b9923d02e.png">
